lighthouse-core/gather/driver.js

```diff
@@ -1487,6 +1487,19 @@ class Driver {
     return this.sendCommand('Network.setExtraHTTPHeaders', {headers});
   }
 
+  /**
+   * @param {LH.Crdp.Network.Cookies|null} cookies key/value pairs of HTTP Cookies.
+   * @return {Promise<void>}
+   */
+  async setCookies(cookies) {
+    if (!cookies) {
+      return;
+    }
+
+    return this.sendCommand('Network.setCookies', {cookies});
```

Comment:
AFAIK when the domain/url/path isn't set it will just use the existing target, because we're running this before we've navigated to the page I'm guessing the common case won't actually send the headers

Reply:
As far as I remember it is executed on "about:blank" page. This is why it is not really set anything, if not specifying either domain or URL. This another +1 for the bigger JSON structure.

Reply:
> Are we going to entirely defer to the cookie definition for when they're included, e.g. for redirects to different origins, third-party iframes, etc? That format will definitely need to be documented, and it seems like it's going to be a huge pain trying to debug when something goes wrong, but I guess maybe folks are already used to that with other tools doing similar things?
> 
> What does WebPageTest do for setting cookies?

I would assume we should rely on browser behavior for the session. A cookie is just a can-opener to test page usability under the specific conditions:
* authorize access to content
* allow access to CDN images
* local show locale-specific content
* perform sticky session
* canary routing for CI

I don't think LH is intended to use like E2E test which might need any specific cookie handling between the tests.

Reply:
@RynatSibahatau WDYT about asserting that we have either a url or domain on all the cookies here?